### PR TITLE
Add test case for pkg/kubelet/kubeletconfig/configsync.go

### PR DESCRIPTION
**Pull request by changshuchao**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by jsturtevant** [comment]
Thanks for adding tests.  Why only the `latestNodeConfigSource` and not other functionality in the `configsync` file?  

It would nice to test the functionality of that function that looks to be a bug in the past: 

https://github.com/kubernetes/kubernetes/blob/442a69c3bdf6fe8e525b05887e57d89db1e2f3a5/pkg/kubelet/kubeletconfig/configsync.go#L227-L231

File: pkg/kubelet/kubeletconfig/configsync_test.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+			NodeName:  "nodeApi",
+			expectErr: fmt.Errorf("failed to cast object from informer's store to Node, can't sync config source for Node %q", "nodeApi"),
+			expect:    nil,
+		},
+	}
+
+	for _, test := range tests {
+		actual, err := latestNodeConfigSource(indexer, test.NodeName)
+		if err == nil {
+			if test.expect == actual || !reflect.DeepEqual(test.expect, actual) {
```
Comment (adisky) [inline pkg/kubelet/kubeletconfig/configsync_test.go:88]: This looks confusing!! Why checking using both `==` and reflect?
 please print expected and actual value in log.

Comment (changshuchao) [reply pkg/kubelet/kubeletconfig/configsync_test.go:88]: ok

**Pull request closed without merging**